### ISSUE TITLE
[Bug] Dynamic FP8 quantization fails due to incorrect tensor shape

### ISSUE THREAD

**Issue by qeternity** [sgl-project/sglang#1178]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [X] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 5. Please use English, otherwise it will be closed.

### Describe the bug

Dynamic FP8 quantization fails with the following error:

```
File "/root/sglang/.venv/lib/python3.11/site-packages/vllm/model_executor/layers/quantization/utils/marlin_utils.py", line 172, in marlin_permute_scales
    s = s.reshape((-1, len(scale_perm_single)))[:, scale_perm_single]
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: shape '[-1, 32]' is invalid for input of size 1
, detoken_init_state: init ok
```

### Reproduction

Load fp16 model with `--quantization fp8`.

### Environment

```
Python: 3.11.6 (main, Apr 10 2024, 17:26:07) [GCC 13.2.0]
CUDA available: True
GPU 0,1: NVIDIA GeForce RTX 3090 Ti
GPU 0,1 Compute Capability: 8.6
CUDA_HOME: /usr/local/cuda-12.5
NVCC: Cuda compilation tools, release 12.5, V12.5.40
CUDA Driver Version: 555.42.02
PyTorch: 2.4.0+cu121
sglang: 0.2.13
flashinfer: 0.1.5+cu124torch2.4
triton: 3.0.0
transformers: 4.44.0
requests: 2.32.3
tqdm: 4.66.5
numpy: 1.26.4
aiohttp: 3.10.4
fastapi: 0.112.1
hf_transfer: 0.1.8
huggingface_hub: 0.24.5
interegular: 0.3.3
packaging: 24.1
PIL: 10.4.0
psutil: 6.0.0
pydantic: 2.8.2
uvicorn: 0.30.6
uvloop: 0.20.0
zmq: 26.1.0
vllm: 0.5.4
multipart: 0.0.9
openai: 1.41.0
anthropic: 0.34.0
NVIDIA Topology:
        GPU0    GPU1    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      PHB     0-11    0               N/A
GPU1    PHB      X      0-11    0               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

ulimit soft: 1024
```

**Comment by intervitens** [comment]
+1, having the same problem

**Comment by halexan** [comment]
3090 not support fp8

**Comment by qeternity** [comment]
It doesn't have native hardware support. But this worked with previous versions just fine, and even with latest I can run static fp8 with no problem.

**Comment by alxiang** [comment]
This was fixed in vllm==0.5.5: https://github.com/vllm-project/vllm/pull/7219

**Comment by qeternity** [comment]
Thanks!
